tests/QueryTest.php

```diff
@@ -401,6 +402,32 @@ public function testWithColumnsSupportsExpressions()
         );
     }
 
+    public function testWithColumnsSupportsAliasedExpressions()
+    {
+        $model = new class () extends User {
+            public function getColumns(): array
+            {
+                return ['username', 'aliased_expr' => new Expression('1')];
+            }
+        };
+
+        $query = (new Query())
+            ->setModel($model)
+            ->withColumns(['aliased_expr', 'username']);
```

Comment:
This may lead to false positives in the future. It's not guaranteed that the call to `withColumns`, with an expression alias of the base table, will trigger this bug. (the test should still succeed if you remove this call) Please use relation columns instead.

Reply:
> It's not guaranteed that the call to `withColumns`, with an expression alias of the base table, will trigger this bug

Calling `withColumns` with an expression alias isn't supposed to trigger the original bug, since it's computed with so many nested built-in function calls :) (flip, filter, keys), but never `array_diff`.

Reply:
Okay, maybe not. But the test still shouldn't succeed if it's removed.

Reply:
It shouldn't succeed now!